CustomCraftSML/Serialization/Entries/CustomFood.cs

```diff
@@ -0,0 +1,298 @@
+                    return imagePath;
+
+                if (this.FoodValue >= this.WaterValue)
+                    return "cake.png";
+                else
+                    return "juice.png";
+            }
+        }
+
+        public CustomFood() : this(TypeName, CustomFoodProperties)
+        {
+        }
+
+        protected CustomFood(string key) : this(key, CustomFoodProperties)
+        {
+        }
+
+        protected CustomFood(string key, ICollection<EmProperty> definitions) : base(key, definitions)
+        {
+            foodValue = (EmProperty<short>)Properties[FoodKey];
+            waterValue = (EmProperty<short>)Properties[WaterKey];
+            decayrate = (EmProperty<float>)Properties[DecayRateKey];
+            allowOverfill = (EmYesNo)Properties[OverfillKey];
+
+            techGroup.Value = TechGroup.Survival;
+            techCategory.DefaultValue = TechCategory.CookedFood;
+            foodModel = (EmProperty<FoodModel>) Properties[FoodModelKey];
+        }
+
+        internal override EmProperty Copy()
+        {
+            return new CustomFood(this.Key, this.CopyDefinitions);
+        }
+
+        public override bool PassesPreValidation()
+        {
+            return base.PassesPreValidation() & ValidateCustomFoodValues();
+        }
+
+        private bool ValidateCustomFoodValues()
+        {
+            if (this.FoodValue < MinValue || this.FoodValue > MaxValue)
+            {
+                QuickLogger.Warning($"{this.Key} entry '{this.ItemID}' from {this.Origin} has {FoodKey} values out of range. Must be between {MinValue} and {MaxValue}. Entry will be discarded.");
+                return false;
+            }
+
+            if (this.WaterValue < MinValue || this.WaterValue > MaxValue)
+            {
+                QuickLogger.Warning($"{this.Key} entry '{this.ItemID}' from {this.Origin} has {WaterKey} values out of range. Must be between {MinValue} and {MaxValue}. Entry will be discarded.");
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override void HandleCustomSprite()
+        {
+            string imagePath = IOPath.Combine(FileLocations.AssetsFolder, $"{this.ItemID}.png");
+
+            if (File.Exists(imagePath))
+            {
+                QuickLogger.Debug($"Custom sprite found in Assets folder for {this.Key} '{this.ItemID}' from {this.Origin}");
+                SpriteHandler.RegisterSprite(this.TechType, ImageUtils.LoadSpriteFromFile(imagePath));
+                return;
+            }
+
+            if (this.SpriteItemID > TechType.None && this.SpriteItemID < TechType.Databox)
+            {
+                QuickLogger.Debug($"{SpriteItemIdKey} '{this.SpriteItemID}' used for {this.Key} '{this.ItemID}' from {this.Origin}");
+                SpriteHandler.RegisterSprite(this.TechType, SpriteManager.Get(this.SpriteItemID));
+                return;
+            }
+
+            // TODO - Handle more custom icons
```

Comment:
We should find a way to include the original set of  CustomFood icons and allow the user to select from these basic recolored icons if they choose.  
If this isn't found either, we should log that we are defaulting the sprite.

We should use custom sprites for the defaults so that custom foods always look different from the original `NutrientBlock` and `FilteredWater` even if no icon options are selected by the user.